boa/controller.py

```diff
@@ -1,37 +1,91 @@
+"""
+###################################
+Controller
+###################################
+
+The Controller class controls the optimization
+
+"""
+from __future__ import annotations
+
 import logging
+import os
 import time
 from pathlib import Path
+from typing import Type
+
+from ax.service.scheduler import Scheduler
 
 from boa.ax_instantiation_utils import get_experiment, get_scheduler
 from boa.runner import WrappedJobRunner
 from boa.storage import scheduler_to_json_file
 from boa.utils import get_dictionary_from_callable
-from boa.wrapper_utils import get_dt_now_as_str
+from boa.wrappers.wrapper import BaseWrapper
+from boa.wrappers.wrapper_utils import get_dt_now_as_str
 
 
 class Controller:
-    def __init__(self, config_path, wrapper):
+    """
+    Controls the instantiation of your :mod:`.wrapper` and the
+    necessary ax objects to start your experiment and control
+    the ax scheduler. Once it sets up your Experiment, it starts
+    the scheduler to have the scheduler run your trials, and then it
+    saves the scheduler to a json file.
+
+    Parameters
+    ----------
+    config_path
+        path to configuration yaml or json file
+    wrapper
+        Your Wrapper subclass of BaseWrapper to be instantiated
+
+    """
+
+    def __init__(self, config_path: os.PathLike | str, wrapper: Type[BaseWrapper]):
         self.config_path = config_path
         self.wrapper = wrapper
 
         self.config = None
 
-    def run(self, append_timestamp, experiment_dir, **kwargs):
-        start = time.time()
+        self.scheduler = None
 
+    def setup(
+        self, append_timestamp: bool = None, experiment_dir: os.PathLike = None, **kwargs
+    ) -> tuple[Scheduler, BaseWrapper]:
+        """
+        Sets up all the classes and objects needed to create the ax Scheduler
+
+        Parameters
+        ----------
+        append_timestamp
+            whether to append the output experiment directory with a timestamp or not
+            (default True)
+        experiment_dir
+            output experiment directory to save the experiment and trials to
+            (defaults to what is specified in the config,
```

Comment:
Probably we need to add some examples/instructions about these options to the configuration guide